### PR TITLE
#11921-6/7 Added NDM and ND Snooping Events

### PR DESCRIPTION
**Pull request by qcorporation**
Parent Ticket:
https://github.com/elastic/integrations/issues/11921

Change Log:
- ND snooping events (840x)
- NDM events (61xx)
- Added docs, pipeline-generated logs, and parsing logic

Test Cases:
validated expected both in generated and real logs fields are parsed properly based upon documentation

### CODE REVIEW

File: packages/hpe_aruba_cx/_dev/build/docs/README.md [hunk 1]
```diff
@@ -709,31 +709,31 @@ Note: Descriptions have not been filled out
...
+| <ip>             | client.ip              |
+| <mac>            | client.mac             |
+| <new_mac>        | client.mac             |
+| <old_mac>        | aruba.ndm.old_mac      |
+| <port>           | aruba.port             |
+| <role>           | aruba.role             |
+| <role1>          | aruba.ndm.role1        |
+| <role2>          | aruba.role             |
+| <time>           | aruba.time.seconds     |
+| <throttle_count> | aruba.throttle_count   |
```
Comment (gogochan) [inline packages/hpe_aruba_cx/_dev/build/docs/README.md:735]: The document doesn't seem to have this field, at which I'm little shocked. I suppose the doc is wrong?

File: packages/hpe_aruba_cx/_dev/build/docs/README.md [hunk 2]
```diff
@@ -709,31 +709,31 @@ Note: Descriptions have not been filled out
...
-| aruba.ndm.vrf       |             |      | aruba.vrf.id                 |
+| Docs Field       | Schema Mapping         |
+|------------------|------------------------|
+| <ip>             | client.ip              |
+| <mac>            | client.mac             |
+| <new_mac>        | client.mac             |
+| <old_mac>        | aruba.ndm.old_mac      |
+| <port>           | aruba.port             |
+| <role>           | aruba.role             |
+| <role1>          | aruba.ndm.role1        |
```
Comment (gogochan) [inline packages/hpe_aruba_cx/_dev/build/docs/README.md:732]: `aruba.ndm.old_role` would mean more than `role1` imo.

File: packages/hpe_aruba_cx/_dev/build/docs/README.md (hunk 1 above)
Comment (gogochan) [reply packages/hpe_aruba_cx/_dev/build/docs/README.md:735]: I see that you got this from the log provided by the customer.

File: packages/hpe_aruba_cx/data_stream/log/elasticsearch/ingest_pipeline/default.yml [hunk 3]
```diff
@@ -1037,7 +1037,77 @@ processors:
...
+      patterns:
+        - "^(Static )?Neighbor %{IP:client.ip} modified on Port %{DATA:aruba.port} and VRF %{DATA:aruba.vrf.id} from mac %{MAC:aruba.ndm.old_mac} to new mac %{MAC:client.mac}"
+  - grok:
+      if: "['6132','6133','6134'].contains(ctx.event?.code)"
+      tag: ndm_event_6132_6133_6134
+      field: "message"
+      description: "Duplicate IP detected from ARP reply | Duplicate IPv6 address detected from Neighbour advertisement | Duplicate IP detected from ARP request"
+      patterns:
+        - "^Duplicate (IPv4|IPv6) address %{IP:client.ip} is detected on port %{DATA:aruba.port} with a MAC address of %{MAC:client.mac}"
+        - "^Throttled %{NUMBER:aruba.throttle_count:long} Messages"
```
Comment (gogochan) [inline packages/hpe_aruba_cx/data_stream/log/elasticsearch/ingest_pipeline/default.yml:1109]: Should we note that this pattern isn't present in the doc?

Comment (qcorporation) [reply packages/hpe_aruba_cx/data_stream/log/elasticsearch/ingest_pipeline/default.yml:1109]: @gogochan 
Here's in the thing, take for example, Event ID: 6132, which has a message of
`Duplicate IPv4 address <ip> is detected on port <port> with a MAC address of <mac>' throttle_count: 40`

The notation of `' throttle_count: 40` and other similar notations for different event types are peppered throughout the online docs. Does that mean we must handle these as alternative messages for the same message ID?

I'm questioning if we might have to go back and do that for all of them.

To your comment, "Should we note this pattern isn't present in the docs?", It is, but we don't handle them for other event types. This is the first example in the actual logs where I've seen this

File: packages/hpe_aruba_cx/_dev/build/docs/README.md (hunk 2 above)
Comment (qcorporation) [reply packages/hpe_aruba_cx/_dev/build/docs/README.md]: thanks for the suggestion, fixed

File: packages/hpe_aruba_cx/data_stream/log/elasticsearch/ingest_pipeline/default.yml (hunk 3 above)
Comment (dwhyrock) [reply packages/hpe_aruba_cx/data_stream/log/elasticsearch/ingest_pipeline/default.yml:1109]: Maybe it depends on whether we anchor the end of the messages with `$` in the grok pattern?

If we do, then the messages with the added `throttle_count: 40` part will fail to parse, correct?  Is it better to fail to parse, then fix the message?  Or to "succeed" (by not anchoring the end) and skip setting `aruba.throttle_count`?

Comment (qcorporation) [reply packages/hpe_aruba_cx/data_stream/log/elasticsearch/ingest_pipeline/default.yml:1109]: @dwhyrock I am slightly confused with this comment 
Within the example logs, they are two distinct messages - here's an example
```
2024-02-21T15:36:53.206545-05:00 8360-Primaire ndmd[884]: Event|6132|LOG_ERR|AMM|1/1|Duplicate IPv4 address 127.0.0.1 is detected on port vlan535 with a MAC address of ab:cd:ef:12:34:56
2023-12-04T10:21:12.768663-05:00 8360-Primaire ndmd[884]: Event|6132|LOG_ERR|AMM|1/1|Throttled 2 Messages
```

Both of them have a message ID of 6132 but they come at different times within the logs

**Review by gogochan — APPROVED** [review]

File: packages/hpe_aruba_cx/data_stream/log/elasticsearch/ingest_pipeline/default.yml (hunk 3 above)
Comment (dwhyrock) [reply packages/hpe_aruba_cx/data_stream/log/elasticsearch/ingest_pipeline/default.yml:1109]: Ah, I misunderstood.  I thought sometimes it had additional text and sometimes it didn't.  I didn't realize they were two separate messages.  Disregard my comment then.

**Review by dwhyrock — APPROVED** [review]

**Pull request merged**